src/include/tsan_suppress.h

```diff
@@ -0,0 +1,302 @@
+ */
+static WT_INLINE void
+__wt_tsan_suppress_sub_uint64(uint64_t *var, uint64_t value)
+{
+    *var -= value;
+}
+
+/*
+ * __wt_tsan_suppress_add_uint64_v --
+ *     TSAN warnings suppression for volatile uint64 add.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_add_uint64_v(volatile uint64_t *var, uint64_t value)
+{
+    *var += value;
+}
+
+/*
+ * __wt_tsan_suppress_store_int64 --
+ *     TSAN warnings suppression for int64 store.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_store_int64(int64_t *vp, int64_t v)
+{
+    __wt_atomic_storei64(vp, v);
+}
+
+/*
+ * __wt_tsan_suppress_add_int64 --
+ *     TSAN warnings suppression for int64 add.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_add_int64(int64_t *var, int64_t value)
+{
+    *var += value;
+}
+
+/*
+ * __wt_tsan_suppress_sub_int64 --
+ *     TSAN warnings suppression for int64 subtract.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_sub_int64(int64_t *var, int64_t value)
+{
+    *var -= value;
+}
+
+/*
+ * __wt_tsan_suppress_load_bool --
+ *     TSAN warnings suppression for bool load.
+ */
+static WT_INLINE bool
+__wt_tsan_suppress_load_bool(bool *vp)
+{
+    return (__wt_atomic_loadbool(vp));
+}
+
+/*
+ * __wt_tsan_suppress_store_bool --
+ *     TSAN warnings suppression for bool store.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_store_bool(bool *vp, bool v)
+{
+    __wt_atomic_storebool(vp, v);
+}
+
+/*
+ * __wt_tsan_suppress_load_bool_v --
+ *     TSAN warnings suppression for volatile bool load.
+ */
+static WT_INLINE bool
+__wt_tsan_suppress_load_bool_v(volatile bool *vp)
+{
+    return (__wt_atomic_loadvbool(vp));
+}
+
+/*
+ * __wt_tsan_suppress_store_bool_v --
+ *     TSAN warnings suppression for volatile bool store.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_store_bool_v(volatile bool *vp, bool v)
+{
+    __wt_atomic_storevbool(vp, v);
+}
+
+/*
+ * __wt_tsan_suppress_load_pointer --
+ *     TSAN warnings suppression for pointer load.
+ */
+static WT_INLINE void *
+__wt_tsan_suppress_load_pointer(void **vp)
+{
+    return (__wt_atomic_load_pointer(vp));
+}
+
+/*
+ * __wt_tsan_suppress_store_pointer --
+ *     TSAN warnings suppression for pointer store.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_store_pointer(void **vp, void *v)
+{
+    __wt_atomic_store_pointer(vp, v);
+}
+
+/*
+ * __wt_tsan_suppress_memcpy --
+ *     TSAN warnings suppression for memory copy.
+ */
+static WT_INLINE void *
+__wt_tsan_suppress_memcpy(void *dest, void *src, size_t count)
+{
+    return (memcpy(dest, src, count));
+}
+
+/*
+ * __wt_tsan_suppress_load_wt_addr_ptr --
+ *     TSAN warnings suppression for WT_ADDR pointer load.
+ */
+static WT_INLINE WT_ADDR *
+__wt_tsan_suppress_load_wt_addr_ptr(void **vp)
+{
+    return (WT_ADDR *)(__wt_atomic_load_pointer(vp));
+}
+
+/*
+ * __wt_tsan_suppress_store_wt_addr_ptr --
+ *     TSAN warnings suppression for WT_ADDR pointer store.
+ */
+static WT_INLINE void
+__wt_tsan_suppress_store_wt_addr_ptr(void **vp, WT_ADDR *v)
```

Comment:
Shouldn't this be const (WT_ADDR **) instead?

Reply:
This one is a bit tricky since `ref->addr` which is passed to this function is `void*` so that's the best what we can do here.